step_start_instance.go

```diff
@@ -31,6 +33,14 @@ func (s *stepStartInstance) Run(state multistep.StateBag) multistep.StepAction {
 		context.LoggerFromContext(ctx).WithField("err", err).Error("couldn't start instance")
 		context.CaptureError(ctx, err)
 
+		jobAbortErr, ok := errors.Cause(err).(workerErrors.JobAbortError)
+		if ok {
+			logWriter := state.Get("logWriter").(LogWriter)
+			logWriter.WriteAndClose([]byte(jobAbortErr.UserFacingErrorMessage()))
+
+			return multistep.ActionHalt
```

Comment:
I think you need to set the state to errored with `buildJob.Finish(FinishStateErrored)` as well, since `stepUpdateState` is "after" `stepStartInstance` in the processor stack, and as such won't get run if we halt here.